src/main/java/org/semver4j/internal/range/processor/XRangeProcessor.java

```diff
@@ -64,9 +61,12 @@ public String process(String range) {
                         compareSign = LT.asString();
                         if (isX(minor)) {
                             major = major + 1;
+                            minor = 0;
```

Comment:
We definitively need test case for this. You could add new test in https://github.com/semver4j/semver4j/blob/e2c265f9f417b0e3d159da9113e695fe4d980752/src/test/java/org/semver4j/SemverTest.java#L851 as an additional arguments around `Comparators` section.

Reply:
Tests added.